frontend/lib/resolution/Resolver.cpp

```diff
@@ -386,21 +399,93 @@ types::QualifiedType Resolver::typeErr(const uast::AstNode* ast,
   return t;
 }
 
-/* llvm::SmallVector<const Scope*> Resolver::gatherReceiverAndParentScopesForType( */
-/*     Context* context, const Type* thisType) { */
-/*   llvm::SmallVector<const Scope*> scopes; */
+/**
+  Find scopes for superclasses of a class.  The passed ID should refer to a
+  Class declaration node.  If not, this function will return an empty vector.
+
+  This function is temporary and should only be used in scopeResolveOnly mode.
```

Comment:
If this function was put on `Resolver` we could then `assert(scopeResolveOnly)` in its body to reduce the likelihood of it being used incorrectly. Up to you though, I don't think it's too likely to be misused as-is.

Reply:
I'll add such an assert to `gatherReceiverAndParentScopesForDeclId` which is the only thing that calls this function.

Reply:
Oops, no, I can't do that, because `gatherReceiverAndParentScopesForDeclId` is `static`. I could make `methodReceiverScopes` assert, though. I'll try that and we can remove the assertion if it causes problems.

Reply:
OK, so tests don't pass with such an assertion. I'll add a TODO comment instead, for now.